geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/partitions/AttributeScheme.scala

```diff
@@ -35,7 +36,15 @@ case class AttributeScheme(attribute: String, index: Int, binding: Class[_]) ext
 
   override def getPartitionName(feature: SimpleFeature): String = {
     val value = feature.getAttribute(index)
-    if (value == null) { "" } else { AttributeIndexKey.typeEncode(value) }
+    if (value == null)
+      return defaultPartition
+    val encodedValue = AttributeIndexKey.typeEncode(value)
+    if (allowedValues.nonEmpty) {
+      if (allowedValues.contains(value))
+        return encodedValue
+      return defaultPartition
+    }
+    encodedValue
```

Comment:
comparison needs to be against encodedValue here i believe. Might be a little clearer to not have explicit returns too (generally i think simple up-front checks like the null check followed by a return are fine, but they shouldn't be sprinkled throughout the method). also, we try to use {} around most statements for clarify.

```suggestion
    if (value == null) {
      return defaultPartition
    }
    val encodedValue = AttributeIndexKey.typeEncode(value)
    if (allowedValues.isEmpty || allowedValues.contains(encodedValue)) {
      encodedValue
    } else {
      defaultPartition
    }
```